docs/src/main/asciidoc/security-openid-connect-multitenancy.adoc

```diff
@@ -1110,6 +1110,75 @@ The default tenant configuration is automatically disabled when `quarkus.oidc.au
 
 Be aware that tenant-specific configurations can also be disabled, for example: `quarkus.oidc.tenant-a.tenant-enabled=false`.
 
+[[programmatic-startup]]
+== Programmatic OIDC start-up for multitenant application
+
+Static OIDC tenants can be created programmatically like in the example below:
+
+[source,java]
+----
+package io.quarkus.it.oidc;
+
+import io.quarkus.oidc.Oidc;
+import io.quarkus.oidc.OidcTenantConfig;
+import jakarta.enterprise.event.Observes;
+
+public class OidcStartup {
+
+    void observe(@Observes Oidc oidc) { <1>
+        oidc.create(OidcTenantConfig.authServerUrl("http://localhost:8180/realms/tenant-one").tenantId("tenant-one").build());
+        oidc.create(OidcTenantConfig.authServerUrl("http://localhost:8180/realms/tenant-two").tenantId("tenant-two").build());
+    }
+
+}
+----
+<1> Static tenants must be created in a synchronous manner from the observer method.
+
+The code above is a programmatic equivalent to the following configuration in the `application.properties` file:
+
+[source,properties]
+----
+quarkus.oidc.tenant-one.auth-server-url=http://localhost:8180/realms/tenant-one
+quarkus.oidc.tenant-two.auth-server-url=http://localhost:8180/realms/tenant-two
+----
+
+You can also create static tenants based on the tenants configured in the `application.properties` file.
```

Comment:
As I said, it does look interesting, but I was thinking about it, and started having some doubts:
* It is a fairly advanced case.
* Here we document accessing a runtime package `OidcConfig`, `namedTenants`, we make it part of the API
* This raises the whole lot of questions: what if the users change other properties available at the builder level

IMHO, it is a bit early to start documenting this option. 
Michal, honestly, there is something interesting here - where we let users customize configured tenants, thanks for keeping coming up with these new ideas, but IMHO it requires more discussions, please don't hesitate to open new issue to record your ideas (this one, and dynamic tenant allocation to match annotations)

Reply:
The point of this example was mostly:

- show you can inject config properties because they are going to be ready (whether it is OidcConfig)
- I suspect that there will be always properties that you know before hand and doesn't change for your use case, so there is no point having them hardcoded in Java. I don't know what is mainstream, but the company I worked in past really hated hardcoding stuff in Java, so I think users would appreciate knowing they can only set _some_ dynamic properties

I'll drop this from documentation for now. I don't think we should block this PR by discussing it too much here, but IMHO we should come back to this, I have added this on my list and will come with it (either in form of issue or discussion)